lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,407 @@
+        detector: -> (value) {
+          case
+          when value.key?(:cli)
+            cli
+          when value.key?(:gradle)
+            gradle
+          when value.key?(:maven)
+            maven
+          else
+            base
+          end
+        }
+      )
+    end
+
+    def self.ci_config_section_name
+      'detekt'
+    end
+
+    def analyzer_name
+      'detekt'
+    end
+
+    def analyze(changes)
+      ensure_runner_config_schema(Schema.runner_config) do |config|
+        delete_unchanged_files changes, only: [".kt", ".kts"]
+
+        check_runner_config(config) do |checked_config|
+          @detekt_config = checked_config
+
+          response = 
+            case
+            when gradle_config
+              run_gradle
+            when maven_config
+              run_maven
+            else
+              run_cli
+            end
+
+          if response.kind_of?(Results::Failure)
+            return response
+          end
+          
+          Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+            response.each do |issue|
+              result.add_issue issue
+            end
+          end
+        end
+      end
+    end
+
+    def analyzer_version
+      unknown_version = "0.0.0"
+
+      @analyzer_version ||=
+        case
+        when gradle_config
+          extract_detekt_version_for_gradle || unknown_version
+        when maven_config
+          extract_detekt_version_for_maven || unknown_version
+        else
+          pom_dir = Pathname(ENV['RUNNER_USER_HOME']) / analyzer_name
+          extract_detekt_version_for_maven(pom_dir) || unknown_version
+        end
+    end
+
+    def detekt_config
+      @detekt_config
+    end
+
+    def cli_config
+      detekt_config[:cli]
+    end
+
+    def gradle_config
+      detekt_config[:gradle]
+    end
+
+    def maven_config
+      detekt_config[:maven]
+    end
+
+    def run_cli
+      report_id = "xml"
+      report_path = Tempfile.new("detekt-report")
+      
+      args = [
+        cli_baseline,
+        cli_config_files,
+        cli_config_resource,
+        cli_disable_default_rulesets,
+        cli_excludes,
+        cli_input,
+        cli_report(report_id, report_path.path)
+      ].flatten.compact
+
+      _stdout, stderr, status = capture3(analyzer_bin, *args)
+
+      # detekt will exit with one of the following exit codes:
+      # 0: detekt ran normally and maxIssues count was not reached in BuildFailureReport.
+      # 1: An unexpected error occurred
+      # 2: detekt ran normally and MaxIssues count was reached in BuildFailureReport.
+      # 3: Invalid detekt configuration file detected.
+      if status.exitstatus == 1
+        trace_writer.error stderr.strip
+        raise stderr.strip
+      elsif status.exitstatus == 3
+        return Results::Failure.new(guid: guid, message: stderr.strip, analyzer: analyzer)
+      else
+        return construct_result(report_id, report_path)
+      end      
```

Comment:
[suggestion] What about this change?

- Handle successful exit codes for Runners (not for detekt).
- Set a Runner's custom message for an invalid config. Because users can confirm `stderr` on their build log page.
- I think we don't need to handle `2`. Because when other detekt exit codes will be added, we should fix this code.

```suggestion
      # detekt has some exit codes.
      # @see https://github.com/arturbosch/detekt/blob/1.6.0/docs/pages/gettingstarted/cli.md
      case status.exitstatus
      when 0, 2
        construct_result(report_id, report_path)
      when 3 # invalid configuration
        Results::Failure.new(guid: guid, message: "Your detekt configuration is invalid", analyzer: analyzer)
      else
        trace_writer.error stderr.strip
        raise stderr
      end      
```